components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/internal/ConsentManagerComponent.java

```diff
@@ -110,6 +120,34 @@ protected void unsetRealmService(RealmService realmService) {
         }
     }
 
+    @Reference(
+            name = "pii.controller",
+            service = org.wso2.carbon.consent.mgt.core.connector.PIIController.class,
+            cardinality = ReferenceCardinality.MULTIPLE,
+            policy = ReferencePolicy.DYNAMIC
+    )
+    protected void setPIIController(PIIController piiController) {
+
+        if (piiController != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("PII Controller is registered in ConsentManager service.");
+            }
+
+            if (piiControllers == null) {
+                piiControllers = new ArrayList<>();
```

Comment:
There can be a race condition. Let's initialize this new array list as a class level